libs/payments/ui/src/lib/actions/updateCart.ts

```diff
@@ -29,11 +24,11 @@ export const updateCartAction = async (
       cartDetails,
     });
   } catch (err) {
-    if (err instanceof CouponErrorExpired) {
+    if (err.constructor.name === 'CouponErrorExpired') {
       return CouponErrorMessageType.Expired;
-    } else if (err instanceof CouponErrorGeneric) {
+    } else if (err.constructor === 'CouponErrorGeneric') {
       return CouponErrorMessageType.Generic;
-    } else if (err instanceof CouponErrorLimitReached) {
+    } else if (err.constructor === 'CouponErrorLimitReached') {
```

Comment:
For some reason `instanceof` wouldn't reliably identify the error as an instance of `CouponErrorExpired`. Not sure why this is the case. Any thoughts?